innovation.game.php

```diff
@@ -7315,6 +7340,52 @@ function dogma($card_id) {
         $this->gamestate->nextState('dogmaEffect');
     }
 
+    function endorse($card_to_endorse_id, $card_to_tuck_id) {
+        // Check that this is the player's turn and that it is a "possible action" at this game state
+        self::checkAction('endorse');
+
+        // Ensure that the player still has an endorse action to use this turn
+        if (self::getGameStateValue('endorse_action_state') == 0) {
+            self::throwInvalidChoiceException();
+        }
+
+        $player_id = self::getActivePlayerId();
+        $card_to_endorse = self::getCardInfo($card_to_endorse_id);
+
+        // Check that the player really has this card on his board
+        if ($card_to_endorse['owner'] != $player_id || $card_to_endorse['location'] != "board") {
+            self::throwInvalidChoiceException();
+        }
+        // Card is not at the top of a stack
+        if (!self::isTopBoardCard($card_to_endorse)) {
+            self::throwInvalidChoiceException();
+        }
+        // City cards and Battleship Yamato do not have a featured icon so they cannot be executed as a dogma effect
+        if ($card_to_endorse['dogma_icon'] == null) {
```

Comment:
https://boardgamegeek.com/thread/1815267/dogma-actions-and-cities

Cities can be endorsed

Reply:
Thanks, fixed!